### ISSUE TITLE
Snippets stop working when in Django template html files

### ISSUE THREAD

**Issue by DylanSatow** [NvChad/NvChad#270]
**Describe the bug**
In a blank html file, when I type something like `div`, I am prompted to expand it to `<div></div>`. However, in a file where I type `{% extends 'layout.html' %}` then close and reopen it, I lose all luasnip prompts. I'm not sure if it's possible to register Django templates as valid html in Lsp.

**To Reproduce**
1. Create file test.html
2. add {% extends 'layout.html' %} at the top
3. Close and reopen

**Desktop (please complete the following information):**
 - Mac OS (M1)
 - Iterm2
 - Most recent version of NVIM

**Additional context**

**Comment by JustSage** [comment]
NvChad is not offering language specific nor framework specific settings, hence you will have to sort this out on your own.

Done a bit of searching for your case regardless and I suggest the following:
1) I'd recommend checking providers and python virtual env by `:checkhealth`. 
2) Assuming you're using `LspInstall` it offers `pyright` for python lsp which is currently the best lsp provider. 
3) Check out [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) & [pyright configuration](https://github.com/microsoft/pyright/blob/main/docs/configuration.md) for further configurations (which are necessary if you're using a framework).
4) There is also a reddit thread for [nvim with django](https://www.reddit.com/r/neovim/comments/o5ieh5/neovim_django_lsp_config/) that's worth taking a look.

Hope this will help you!

**Comment by siduck76** [comment]
I agree with @JustSage , would be better to make an issue in luasnip I think.

**Comment by shaeinst** [comment]
@DylanSatow, when you create a blank html file, neovim detects it as a `html` file but when you adds `{% extends 'layout.html' %}` to that file and reopen it, it get detected as `htmldjango` file. that's why LuaSnip didn't load the snippets. to fix this you have to extend htmldjango filetype to html
```lua
-- enable html snippets in Django (htmldjango)
require("luasnip").filetype_extend("htmldjango", {"html"})
```
and if you also want treesitter to work
```lua
require('nvim-treesitter.parsers').filetype_to_parsername.htmldjango = "html"
```
----
@siduck, one thing i missing though is `autotag`. i couldn't make [windwp/nvim-ts-autotag](https://github.com/windwp/nvim-ts-autotag) work with `htmldjango`. any idea?

**Comment by shaeinst** [comment]
sorry, it's actually working.

**Comment by rob32** [comment]
Sorry in advance for going a bit off-topic.
Does anyone here have a tip on how to get TailwindCSS running in `djangohtml`? @shaeinst 

Parts of my custom config: 

`require("luasnip").filetype_extend("htmldjango", { "html" })`

and

```
lspconfig.tailwindcss.setup {
  on_attach = on_attach,
  on_init = on_init,
  capabilities = capabilities,
  filetypes = { "html", "css", "scss", "javascript", "javascriptreact", "typescript", "typescriptreact", "htmldjango", },
  root_dir = lspconfig.util.root_pattern("tailwind.config.js", "tailwind.config.cjs", "tailwind.config.ts", "tailwind.config.tsx", "tailwind.config.scss", "tailwind.config.css"),
}
```

Tailwindcss is attached to the buffer according to `LspInfo` and with _filetype = html_ it works strangely...

I'm using NvChad 2.5 :rocket: :+1:

**Comment by siduck** [comment]
@rob32 check https://www.reddit.com/r/neovim/comments/1bhsoyp/tailwindcss_and_htmldjango/

does it work without nvchad?

**Comment by rob32** [comment]
@siduck Hey, Thanks for the quick reply <3 

Yes, that was me^^

No, unfortunately not. I've really tried everything. Whether with my neovim config with zero-lsp/cmp or lspconfig and cmp, or even with a fresh setup with LazyVim and the tailwindExtra.

The only good thing is that I'm definitely switching to NvChad!
I like it very much. Also the structure and the way of the custom settings.

I really hope that I can get tailwindcss working. 

Would be very grateful for any tips or hints. I am spending so many hours on this now.

**Comment by siduck** [comment]
@rob32  i'm clueless with this! you should make an issue on lspconfig repo, try asking in our discord server too

**Comment by AlixShahid** [comment]
@rob32 
Actually, I am rewriting my [distro](https://github.com/Abstract-IDE/Abstract); it's almost done. Tailwind CSS is working out of the box. The only problem I am facing is that every installed LSP gets attached to the htmldjango file buffer. I think I am missing something. I don't believe the issue lies within your LSP configuration , but rather with your nvim-cmp itself. I don't know how NvChad integrated nvim-cmp, luasnip, mason, and lspconfig. You can take a look at the configs below and figure it out yourself. I am a little busy these days.

[nvim-cmp](https://github.com/Abstract-IDE/Abstract/blob/rewrite/lua/abstract/plugins/cmp.lua)
[LuaSnip](https://github.com/Abstract-IDE/Abstract/blob/rewrite/lua/abstract/plugins/LuaSnip.lua)
[nvim-lspconfig](https://github.com/Abstract-IDE/Abstract/blob/rewrite/lua/abstract/plugins/lspconfig.lua)
[mason](https://github.com/Abstract-IDE/Abstract/blob/rewrite/lua/abstract/plugins/mason.lua)
[mason-lspconfig](https://github.com/Abstract-IDE/Abstract/blob/rewrite/lua/abstract/plugins/mason-lspconfig.lua)
